card-game/src/components/CardSectionLayout.jsx

```diff
@@ -0,0 +1,91 @@
+import styled from 'styled-components';
+import Button from '../components/Button';
+import Card from '../components/Card';
+import { monkeyData } from '../constant/cardData';
+import { useState, useMemo } from 'react';
+
+//배열 섞는 함수
+function shuffle(array) {
+  array.sort(() => Math.random() - 0.5);
+}
+
+
+export default function CardSectionLayout({all, setAll, correct, setCorrect, setSuccess}) {
+
+  const [flippedList,setFlippedList] = useState([]);   // 완성된 카드 배열
+
+  // 난이도가 변경될 때마다 카드 배열 재조합/재정렬
+  const monkeyList = useMemo(() => {
+    return arrayCard(monkeyData);
+  }, [all])
+
+  const monkeyComponentList = monkeyList.map((monkey) => 
+    <Card 
+      key={monkey.id} 
+      idx={monkey.id} 
+      src={monkey.src} 
+      correct={correct} setCorrect={setCorrect} 
+      all={all} 
+      flippedList={flippedList} setFlippedList={setFlippedList} 
+      setSuccess={setSuccess}/>);
```

Comment:
여기서도 구조분해할당을 써보면 어떨까용?
```suggestion
  const monkeyComponentList = monkeyList.map(({id, src}) => 
    <Card 
      key={id} 
      idx={id} 
      src={src} 
      correct={correct} setCorrect={setCorrect} 
      all={all} 
      flippedList={flippedList} setFlippedList={setFlippedList} 
      setSuccess={setSuccess}/>);
```
이렇게 사용해볼 수 있어요! 만일 필드가 엄청 많아지면 오히려 더 복잡해질 수도 있지만, 이렇게 분해해두면 값을 불러올 때 굳이 . 찍어서 안 가져와도 되니까 더 편리할 것이라고 생각합니다!